### PR TITLE
Hashing function in contract env.

### LINKED ISSUE #180: Hashing function in contract_env.
Instead of keccak, use backend's default hashing algorithm, provided by contract_env.
https://github.com/odradev/odra/blob/c734dca1416de7e8442a78cfac8cfb63020693cb/modules/src/mock/access_control.rs#L70

### PR DESCRIPTION
**Pull request by kubaplas**
Closes #180

### CODE REVIEW

File: odra-casper/livenet/src/contract_env.rs [hunk 1]
```diff
@@ -77,3 +77,8 @@ pub fn native_token_metadata() -> NativeTokenMetadata {
 pub fn verify_signature(_: &Bytes, _: &Bytes, _: &PublicKey) -> bool {
     unimplemented!()
 }
+
+/// Creates a hash of the given input. Uses default hash for given backend.
+pub fn hash<T: AsRef<[u8]>>(_: T) -> Bytes {
+    unimplemented!()
```
Comment (zie1ony) [inline odra-casper/livenet/src/contract_env.rs]: `casper_client.rs` file has blake2 implementation you can reuse.

**Review by zie1ony — CHANGES_REQUESTED** [review]

File: mock-vm/backend/src/contract_env.rs [hunk 2]
```diff
@@ -124,3 +126,11 @@ pub fn verify_signature(message: &Bytes, signature: &Bytes, public_key: &PublicK
     let mock_signature_bytes = Bytes::from(message);
     mock_signature_bytes == *signature
 }
+
+/// Creates a hash of the given input. Uses default hash for given backend.
+pub fn hash<T: AsRef<[u8]>>(input: T) -> Bytes {
```
Comment (kpob) [inline mock-vm/backend/src/contract_env.rs]: I wouldn't wrap the bytes. Most likely, a user will use the result in some further processing, so having raw bytes is a more flexible approach.

**Review by kpob — CHANGES_REQUESTED** [review]

File: odra-casper/livenet/src/contract_env.rs (hunk 1 above)
Comment (kubaplas) [reply odra-casper/livenet/src/contract_env.rs]: Fixed

File: mock-vm/backend/src/contract_env.rs (hunk 2 above)
Comment (kubaplas) [reply mock-vm/backend/src/contract_env.rs]: Fixed

**Review by kpob — APPROVED** [review]

**Review by zie1ony — APPROVED** [review]

**Pull request merged**